frontend/www/js/omegaup/components/course/Statistics.vue

```diff
@@ -0,0 +1,197 @@
+      <div class="d-flex justify-content-center">
+        <select v-model="selected" class="text-center">
+          <option v-for="option in options" v-bind:value="option.value">{{
+            option.text
+          }}</option>
+        </select>
+      </div>
+      <highcharts v-bind:options="selected"></highcharts>
+    </div>
+  </div>
+  <!-- panel -->
+</template>
+
+<script lang="ts">
+import { Chart } from 'highcharts-vue';
+import { Vue, Component, Prop, Watch } from 'vue-property-decorator';
+import { types } from '../../api_types';
+import T from '../../lang';
+import * as ui from '../../ui';
+@Component({
+  components: {
+    highcharts: Chart,
+  },
+})
+export default class Statistics extends Vue {
+  @Prop() course!: types.CourseDetails;
+  @Prop() problemStats!: types.CourseProblemStatistics[];
+  T = T;
+  //chart options
+  selected = this.varianceChartOptions;
+  options = [
+    { value: this.varianceChartOptions, text: T.courseStatisticsVariance },
+    { value: this.averageChartOptions, text: T.courseStatisticsAverageScore },
+    {
+      value: this.highScoreChartOptions,
+      text: T.courseStatisticsStudentsAbove,
+    },
+    {
+      value: this.lowScoreChartOptions,
+      text: T.courseStatisticsStudentsScored,
+    },
+    { value: this.minimumChartOptions, text: T.courseStatisticsMinimumScore },
+    { value: this.maximumChartOptions, text: T.courseStatisticsMaximumScore },
+  ];
+  //get chart options
+  get varianceChartOptions() {
+    return this.createChartOptions(
+      T.courseStatisticsScoreVariance,
+      '{y}',
+      T.courseStatisticsVariance,
+      this.getStatistic('variance'),
+      this.maxVariance,
+      this.problems,
+    );
+  }
+  get averageChartOptions() {
+    return this.createChartOptions(
+      T.courseStatisticsAverageScore,
+      '{y}',
+      T.wordsScore,
+      this.getStatistic('average'),
+      this.maxPoints,
+      this.problems,
+    );
+  }
+  get highScoreChartOptions() {
+    return this.createChartOptions(
+      T.courseStatisticsStudentsAbove,
+      '{y} %',
+      T.wordsPercentage,
+      this.getStatistic('high_score_percentage'),
+      100,
+      this.problems,
+    );
+  }
+  get lowScoreChartOptions() {
+    return this.createChartOptions(
+      T.courseStatisticsStudentsScored,
+      '{y} %',
+      T.wordsPercentage,
+      this.getStatistic('low_score_percentage'),
+      100,
+      this.problems,
+    );
+  }
+  get minimumChartOptions() {
+    return this.createChartOptions(
+      T.courseStatisticsMinimumScore,
+      '{y}',
+      T.wordsScore,
+      this.getStatistic('minimum'),
+      this.maxPoints,
+      this.problems,
+    );
+  }
+  get maximumChartOptions() {
+    return this.createChartOptions(
+      T.courseStatisticsMaximumScore,
+      '{y}',
+      T.wordsScore,
+      this.getStatistic('maximum'),
+      this.maxPoints,
+      this.problems,
+    );
+  }
+  //helper functions
+  get problems() {
+    let problems = [];
```

Comment:
minor nit: whenever a variable won't be reassigned, the use of [`const`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/const) is preferred over `let`, since that more clearly marks the variable as holding the same value throughout the function:

```suggestion
    const problems = [];
```